src/main/java/lt/mariusworks/work1/services/SchoolService.java

```diff
@@ -0,0 +1,34 @@
+package lt.mariusworks.work1.services;
+
+import lombok.RequiredArgsConstructor;
+import lt.mariusworks.work1.entities.School;
+import lt.mariusworks.work1.entities.Student;
+import lt.mariusworks.work1.repositories.SchoolRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+@RequiredArgsConstructor
+@Service
+public class SchoolService {
+    private final SchoolRepository schoolRepository;
+
+
+    public List<School> getAllSchools() {
+        return this.schoolRepository.findAll();
+    }
+
+    public School getSchoolByIdOrThrow(Long id) {
+        return this.schoolRepository.findById(id).orElseThrow();
+    }
+
+    public Student getStudentBySchoolAndId(Long schoolId, Long studentId) {
+        for (Student student : this.schoolRepository.findById(schoolId).orElseThrow().getStudents()) {
```

Comment:
perpanaudojam jau turimą kodą:

`for (Student student : this. getSchoolByIdOrThrow(schoolId).getStudents()) {`